velox/docs/configs.rst

```diff
@@ -4,230 +4,172 @@ Configuration properties
+   * - order_by_spill_memory_threshold
+     - integer
+     - 0
+     - Maximum amount of memory in bytes that an order by can use before spilling. 0 means unlimited.
+   * - spillable-reservation-growth-pct
+     - integer
+     - 25
+     - The spillable memory reservation growth percentage of the current memory reservation size. Suppose a growth
+       percentage of N and the current memory reservation size of M, the next memory reservation size will be
+       M * (1 + N / 100). After growing the memory reservation K times, the memory reservation size will be
+       M * (1 + N / 100) ^ K. Hence the memory reservation grows along a series of powers of (1 + N / 100).
+       If the memory reservation fails, it starts spilling.
+   * - max-spill-level
+     - integer
+     - 4
+     - The maximum allowed spilling level with zero being the initial spilling level. Applies to hash join build
+       spilling which might use recursive spilling when the build table is very large. -1 means unlimited.
+       In this case an extremely large query might run out of spilling partition bits. The max spill level
+       can be used to prevent a query from using too much io and cpu resources.
+   * - max-spill-file-size
+     - integer
+     - 0
+     - The maximum allowed spill file size. Zero means unlimited.
+   * - min-spill-run-size
+     - integer
+     - 256MB
+     - The minimum spill run size (bytes) limit used to select partitions for spilling. The spiller tries to spill a
+       previously spilled partitions if its data size exceeds this limit, otherwise it spills the partition with most data.
+       If the limit is zero, then the spiller always spills a previously spilled partition if it has any data. This is
+       to avoid spill from a partition with a small amount of data which might result in generating too many small
+       spilled files.
 
 Hive Connector
------------------------------
-
-``max_partitions_per_writers``
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
-    * **Type:** ``integer``
-    * **Default value:** ``100``
-
-Maximum number of partitions per a single table writer instance.
-
-``insert_existing_partitions_behavior``
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
-    * **Type:** ``string``
-    * **Allowed values:** ``OVERWRITE``, ``ERROR``
-    * **Default value:** ``ERROR``
-
-The behavior on insert existing partitions. This property only derives
-the update mode field of the table writer operator output. ``OVERWRITE``
-sets the update mode to indicate overwriting a partition if exists.
-``ERROR`` sets the update mode to indicate error throwing if writing
-to an existing partition.
-
-``immutable_partitions``
-^^^^^^^^^^^^^^^^^^^^^^^^
-
-    * **Type:** ``bool``
-    * **Default value:** ``false``
-
-True if appending data to an existing unpartitioned table is allowed.
-Currently this configuration does not support appending to existing partitions.
-
+--------------
+``Generic Configuration``
```

Comment:
Should we remove Generic Configuration ?

Reply:
Yes, we should. Removed.